tools/integration_tests/implicit_dir/delete_test.go

```diff
@@ -0,0 +1,140 @@
+// testBucket/implicitDirectory/fileInImplicitDir1                               -- File
+// testBucket/implicitDirectory/implicitSubDirectory                             -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2          -- File
+func TestDeleteNonEmptyImplicitSubDir(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectory()
+
+	subDirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory, implicit_and_explicit_dir_setup.ImplicitSubDirectory)
+
+	implicit_and_explicit_dir_setup.RemoveAndCheckIfDirIsDeleted(subDirPath, implicit_and_explicit_dir_setup.ImplicitSubDirectory, t)
+}
+
+// Directory Structure
+// testBucket/implicitDirectory                                                                    -- Dir
+// testBucket/implicitDirectory/explicitDirInImplicitDir                                           -- Dir
+// testBucket/implicitDirectory/explicitDirInImplicitDir/fileInExplicitDirInImplicitDir            -- File
+// testBucket/implicitDirectory/fileInImplicitDir1                                                 -- File
+// testBucket/implicitDirectory/implicitSubDirectory                                               -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2                            -- File
+func TestDeleteImplicitDirWithExplicitSubDir(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectory()
+
+	explicitDirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory, ExplicitDirInImplicitDir)
+
+	operations.CreateDirectoryWithNFiles(NumberOfFilesInExplicitDirInImplicitDir, explicitDirPath, PrefixFileInExplicitDirInImplicitDir, t)
+
+	dirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory)
+
+	implicit_and_explicit_dir_setup.RemoveAndCheckIfDirIsDeleted(dirPath, implicit_and_explicit_dir_setup.ImplicitDirectory, t)
+}
+
+// Directory Structure
+// testBucket/implicitDirectory                                                                                         -- Dir
+// testBucket/implicitDirectory/fileInImplicitDir1                                                                      -- File
+// testBucket/implicitDirectory/implicitSubDirectory                                                                    -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2                                                 -- File
+// testBucket/implicitDirectory/implicitSubDirectory/explicitDirInImplicitDir                                           -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/explicitDirInImplicitDir/fileInExplicitDirInImplicitDir            -- File
+func TestDeleteImplicitDirWithExplicitSubDirInImplicitSubDir(t *testing.T) {
```

Comment:
Please rename it to - TestDeleteImplicitDirWithImplicitSubDirContainingExplicitDir.